smoke-test/src/test/java/org/opennms/smoketest/DaemonConfigPageIT.java

```diff
@@ -0,0 +1,258 @@
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import com.google.common.base.Predicate;
+
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+
+public class DaemonConfigPageIT extends OpenNMSSeleniumTestCase {
+    @Rule
+    public Timeout timeout = new Timeout(10, TimeUnit.MINUTES);
+
+    @Before
+    public void setUp() {
+        RestAssured.baseURI = getBaseUrl();
+        RestAssured.port = getServerHttpPort();
+        RestAssured.basePath = "/opennms/admin/daemons/index.jsp";
+        RestAssured.authentication = preemptive().basic(BASIC_AUTH_USERNAME, BASIC_AUTH_PASSWORD);
+    }
+
+    @After
+    public void tearDown() {
+        RestAssured.reset();
+    }
+
+    @Test
+    public void verifyDaemonListIsShown() {
+        given().get().then()
+                .assertThat().statusCode(200)
+                .assertThat().contentType(ContentType.HTML);
+
+        DaemonReloadPage page = new DaemonReloadPage().open();
+        Assert.assertThat(page.getDaemonRows(), hasSize(30));
+
+        Daemon eventd = new Daemon(page, "eventd");
+        Daemon alarmd = new Daemon(page, "alarmd");
+        Daemon snmppoller = new Daemon(page, "snmppoller");
+        Daemon ticketer = new Daemon(page, "ticketer");
+
+        Assert.assertThat(eventd.getName(), is("eventd"));
+        Assert.assertThat(eventd.getStatus(), is("running"));
+        Assert.assertThat(eventd.isReloadable(), is(true));
+
+        Assert.assertThat(alarmd.getName(), is("alarmd"));
+        Assert.assertThat(alarmd.getStatus(), is("running"));
+        Assert.assertThat(alarmd.isReloadable(), is(true));
+
+        Assert.assertThat(snmppoller.getName(), is("snmppoller"));
+        Assert.assertThat(snmppoller.getStatus(), is("not running"));
+        Assert.assertThat(snmppoller.isReloadable(), is(false));
+
+        Assert.assertThat(ticketer.getName(), is("ticketer"));
+        Assert.assertThat(ticketer.getStatus(), is("running"));
+        Assert.assertThat(ticketer.isReloadable(), is(false));
+
+
+        Assert.assertThat(eventd.reload(5), is("Success"));
+        Assert.assertThat(eventd.getCurlString(), is(getExceptedCurlStringForDaemonName(eventd.getName())));
+
+        Assert.assertThat(alarmd.reload(5), is("Success"));
+        Assert.assertThat(alarmd.getCurlString(), is(getExceptedCurlStringForDaemonName(alarmd.getName())));
+    }
+
+    private String getExceptedCurlStringForDaemonName(String name){
+        return String.format("curl --request POST -u USERNAME:PASSWORD -url %sopennms/rest/daemons/reload/%s/", getBaseUrl(), name);
+    }
+
+    private class Daemon {
+        private final DaemonReloadPage page;
+        private final String daemonName;
+
+        public Daemon(DaemonReloadPage page, String daemonName) {
+            this.page = page;
```

Comment:
This looks weird, as this seems to be an enforced relationship, but there is no need to. Probably better to make the `page.getRowForDaemonName(this.daemonName)` and `page.getPopupPreElementForDaemonName(this.daemonName)` methods utility classes, as they don't require anything from the page